src/main/java/gov/usds/case_issues/services/SneakyCaseService.java

```diff
@@ -0,0 +1,336 @@
+			DateRange caseCreationRange, Optional<String> snoozeReason, int size) {
+		return getCases(CaseSnoozeFilter.SNOOZED, caseManagementSystemTag, caseTypeTag, SNOOZE_SORT, Optional.ofNullable(receiptNumber), size,
+				Optional.ofNullable(caseCreationRange),
+				snoozeReason, Collections.emptyMap(), Optional.empty());
+	}
+
+	@Override
+	public List<CaseSummary> getPreviouslySnoozedCases(String caseManagementSystemTag, String caseTypeTag,
+			String receiptNumber, DateRange caseCreationRange, int size) {
+		return getCases(CaseSnoozeFilter.ALARMED, caseManagementSystemTag, caseTypeTag, DEFAULT_SORT, Optional.ofNullable(receiptNumber), size,
+				Optional.ofNullable(caseCreationRange),
+				Optional.empty(), Collections.emptyMap(), Optional.empty());
+	}
+
+	public List<CaseSummary> getCases(CaseSnoozeFilter queryFilter, String caseManagementSystemTag, String caseTypeTag, 
+			Sort sortOrder, Optional<String> pageReference, int pageSize,
+			@NotNull Optional<DateRange> caseCreationRange,
+			@NotNull Optional<String> snoozeReason,
+			@NotNull Map<String, Object> fieldFilter,
+			@NotNull Optional<AttachmentRequest> attachmentFilter
+			) {
+		if (sortOrder == null) {
+			sortOrder = DEFAULT_SORT;
+		}
+		Specification<SneakyViewEntity> mainSpec = commonSpec(caseManagementSystemTag, caseTypeTag, sortOrder,
+				pageReference, caseCreationRange);
+		if (queryFilter != null) {
+			mainSpec = mainSpec.and(caseCategorySpec(queryFilter));
+		}
+		if (snoozeReason.isPresent()) {
+			mainSpec = mainSpec.and((root, query, cb) -> cb.equal(root.get("snoozeReason"), snoozeReason.get()));
+		}
+		if (fieldFilter != null && fieldFilter.size() > 0) {
+			mainSpec = mainSpec.and(caseExtraDataSpec(fieldFilter));
+		}
+		if (attachmentFilter.isPresent()) {
+			mainSpec = mainSpec.and(attachmentSpec(attachmentFilter.get()));
+		}
+		return wrapFetched(mainSpec, PageRequest.of(0, pageSize, sortOrder));
+	}
+
+	private Specification<SneakyViewEntity> attachmentSpec(AttachmentRequest attachmentRequest) {
+		return (root, query, cb) -> {
+			Subquery<CaseAttachmentAssociation> sq = query.subquery(CaseAttachmentAssociation.class);
+			List<Predicate> conjunction = new ArrayList<>();
+			Root<CaseAttachmentAssociation> aRoot = sq.from(CaseAttachmentAssociation.class);
+			sq.select(aRoot);
+			Path<CaseAttachment> aPath = aRoot.get("attachment");
+			conjunction.add(cb.equal(aPath.get("attachmentType"), cb.literal(attachmentRequest.getNoteType())));
+			if (attachmentRequest.getSubtype() != null) {
+				conjunction.add(cb.equal(aPath.get("attachmentSubtype").get("externalId"), cb.literal(attachmentRequest.getSubtype())));
+			}
+			if (attachmentRequest.getContent() != null) {
+				conjunction.add(cb.equal(aPath.get("content"), cb.literal(attachmentRequest.getContent())));
+
+			}
+			conjunction.add(cb.equal(aRoot.get("snooze").get("snoozeCase").get("internalId"), root.get("internalId")));
```

Comment:
This is a good place to post this reminder: https://vladmihalcea.com/hibernate-facts-always-check-criteria-api-sql-queries/

```sql
  and (
            exists (
                select
                    caseattach1_.internal_id 
                from
                    case_issue_main.case_attachment_association caseattach1_ cross 
                join
                    case_issue_main.case_attachment caseattach2_ cross 
                join
                    case_issue_main.attachment_subtype attachment4_ cross 
                join
                    case_issue_main.case_snooze casesnooze5_ 
                where
                    caseattach1_.attachment_internal_id=caseattach2_.internal_id 
                    and caseattach2_.attachment_subtype_internal_id=attachment4_.internal_id 
                    and caseattach1_.snooze_internal_id=casesnooze5_.internal_id 
                    and caseattach2_.attachment_type=? 
                    and attachment4_.external_id=? 
                    and casesnooze5_.snooze_case_internal_id=sneakyview0_.internal_id
            )
        ) 
```

That's probably not the most efficient query we could have for this, though I'd want to look and see if it's actually as inefficient as it looks or if pg has quietly optimized that series of cross joins into the same query plan that a more modern spelling would give.